### PR TITLE
ENH make compatible with ngmix v2, add tests

### PR DESCRIPTION
**Pull request by beckermr**
This PR makes the code compatible with ngmix v2.

### CODE REVIEW

File: eastlake/steps/newish_metacal/metacal/metacal_fitter.py [hunk 1]
```diff
@@ -159,7 +138,73 @@ def _do_all_metacal(self, mbobs_list, data=None):
         output = eu.numpy_util.combine_arrlist(datalist)
         return output
 
-    def _do_one_metacal(self, mbobs):
+    def _do_one_metacal_ngmixv2(self, mbobs):
```
Comment (beckermr) [inline eastlake/steps/newish_metacal/metacal/metacal_fitter.py:141]: @esheldon can you check this method over? It works locally on my machine just fine, but I want to make sure I have  things set reasonably.

Comment (esheldon) [reply eastlake/steps/newish_metacal/metacal/metacal_fitter.py:141]: Our priors are usually not constraining on T and flux, so PriorGuesser might  not work well

File: eastlake/steps/newish_metacal/metacal/metacal_fitter.py [hunk 2]
```diff
@@ -159,7 +138,73 @@ def _do_all_metacal(self, mbobs_list, data=None):
...
+        gm = GaussMom(1.2).go(mbobs[0][0])
+        if gm['flags'] == 0:
+            flux_guess = gm['flux']
+            Tguess = gm['T']
+            guesser = TFluxAndPriorGuesser(
+                rng=self.rng, T=Tguess, flux=flux_guess, prior=self.metacal_prior,
+            )
+        else:
+            guesser = PriorGuesser(self.metacal_prior)
+        psf_guesser = SimplePSFGuesser(rng=self.rng)
```
Comment (esheldon) [inline eastlake/steps/newish_metacal/metacal/metacal_fitter.py:151]: You can send `guess_from_moms=True` to make guesses for the psf using GaussMom

File: eastlake/steps/newish_metacal/metacal/metacal_fitter.py [hunk 3]
```diff
@@ -159,7 +138,73 @@ def _do_all_metacal(self, mbobs_list, data=None):
...
+            guesser=psf_guesser,
+            ntry=self['metacal']['psf']['ntry'],
+        )
+        runner = Runner(
+            fitter=fitter,
+            guesser=guesser,
+            ntry=self['metacal']['max_pars']['ntry'],
+        )
+
+        boot = MetacalBootstrapper(
```
Comment (esheldon) [inline eastlake/steps/newish_metacal/metacal/metacal_fitter.py:174]: Is to run standard metacal as an alternative to metadetect?

Comment (esheldon) [reply eastlake/steps/newish_metacal/metacal/metacal_fitter.py:174]: Note that ` BootPSFFailure` exception can be raised.

File: eastlake/steps/newish_metacal/metacal/metacal_fitter.py (hunk 1 above)
Comment (esheldon) [reply eastlake/steps/newish_metacal/metacal/metacal_fitter.py:141]: I recall in fitvd I fell back on some factor of the PST T value when moments failed.

File: eastlake/steps/newish_metacal/metacal/metacal_fitter.py (hunk 3 above)
Comment (beckermr) [reply eastlake/steps/newish_metacal/metacal/metacal_fitter.py:174]: Yes. This code is used to check the sims when we have grid around.

File: eastlake/steps/newish_metacal/metacal/metacal_fitter.py (hunk 1 above)
Comment (beckermr) [reply eastlake/steps/newish_metacal/metacal/metacal_fitter.py:141]: Good idea!

File: eastlake/steps/newish_metacal/metacal/metacal_fitter.py [hunk 4]
```diff
@@ -125,14 +112,14 @@ def _do_all_metacal(self, mbobs_list, data=None):
...
+                    if NGMIX_V2:
+                        res = self._do_one_metacal_ngmixv2(mbobs)
                     else:
-                        res = boot.get_metacal_result()
+                        boot = self._do_one_metacal_ngmixv1(mbobs)
+                        if isinstance(boot, dict):
+                            res = boot
+                        else:
+                            res = boot.get_metacal_result()
                 except (BootPSFFailure, BootGalFailure) as err:
```
Comment (beckermr) [inline eastlake/steps/newish_metacal/metacal/metacal_fitter.py:123]: I am catching any errors raised from the fitters here, including `BootPSFFailure`.

File: eastlake/steps/newish_metacal/metacal/metacal_fitter.py (hunk 3 above)
Comment (beckermr) [reply eastlake/steps/newish_metacal/metacal/metacal_fitter.py:174]: Errors are caught by the caller of this method.

**Pull request merged**